containers/src/Data/Map/Internal.hs

```diff
@@ -3365,16 +3365,16 @@ foldlWithKey' f z = go z
       in go (f z'' kx x) r
 {-# INLINE foldlWithKey' #-}
 
--- | /O(n)/. Monadic variant of 'foldlWithKey\''.
-foldlWithKeyM' :: Monad m => (a -> k -> b -> m a) -> a -> Map k b -> m a
-foldlWithKeyM' f z = go z
+-- | /O(n)/. Monadic variant of 'foldlWithKey'.
+foldlWithKeyM :: Monad m => (a -> k -> b -> m a) -> a -> Map k b -> m a
+foldlWithKeyM f z = go z
   where
-    go !z' Tip             = return z'
+    go z' Tip             = return z'
     go z' (Bin _ kx x l r) = do
       !z'' <- go z' l
```

Comment:
This is still strict. I guess it would be okay to have both a lazy and strict version if you want, though it seems a bit redundant.

Reply:
Sorry, that was an oversight. This should be resolved in the latest commit.